ExampleMod/Content/Items/Consumables/PlanteraItem.cs

```diff
@@ -6,13 +6,23 @@
 
 namespace ExampleMod.Content.Items.Consumables
 {
+	//This is the Item used to summon a boss, in this case the vanilla Plantera boss.
 	public class PlanteraItem : ModItem
 	{
 		public override void SetStaticDefaults() {
 			DisplayName.SetDefault("Plantera");
 			Tooltip.SetDefault("The wrath of the jungle");
 			CreativeItemSacrificesCatalog.Instance.SacrificeCountNeededByItemId[Type] = 3;
-			ItemID.Sets.SortingPriorityBossSpawns[Item.type] = 12; // This helps sort inventory know that this is a boss summoning Item.
+			ItemID.Sets.SortingPriorityBossSpawns[Type] = 12; // This helps sort inventory know that this is a boss summoning Item.
+
+			//This is set to true for all NPCs that can be summoned via an Item (calling NPC.SpawnOnPlayer). If this is for a modded boss,
+			//write this in the bosses file instead
+			NPCID.Sets.MPAllowedEnemies[NPCID.Plantera] = true;
+		}
+
+		public override void Unload() {
+			//If the mod is disabled, the flag should be reset again to retain the original state of the game
+			NPCID.Sets.MPAllowedEnemies[NPCID.Plantera] = false;
```

Comment:
@Mirsario should tML be resetting this array? I guess we don't revert the effects of GlobalX on Sets arrays normally. 
Wait.. we re-run the static constructors which should reset them yes?

Reply:
@direwolf420 please test if this is required.

Reply:
Yes, all sets get reset, unloading isn't needed.

Reply:
Yes, just tested, printing false when ExampleMod is unloaded again, will fix